p/scripts/category.js

```diff
@@ -90,31 +102,35 @@ function init_draggable() {
 				// HACK because dragleave is triggered when hovering children!
 				return;
 			}
-			li.classList.remove('drag-hover');
+			ul_dropzone.classList.remove('drag-hover');
 		}
 	};
 
 	dropSection.ondragover = function (ev) {
 		const li = ev.target.closest ? ev.target.closest(dropzone) : null;
 		if (li) {
+			li.closest('ul').classList.remove('drag-drop');
 			ev.dataTransfer.dropEffect = 'move';
 			return false;
 		}
 	};
 
 	dropSection.ondrop = function (ev) {
-		const li = ev.target.closest ? ev.target.closest(dropzone) : null;
-		if (li) {
+		const ul_dropzone = ev.target.closest ? ev.target.closest(dropzone) : null;
+		if (ul_dropzone) {
 			loading = true;
 
 			const req = new XMLHttpRequest();
 			req.open('POST', './?c=feed&a=move', true);
 			req.responseType = 'json';
 			req.onload = function (e) {
 				if (this.status == 200) {
-					li.insertAdjacentHTML('afterend', dragHtml);
-					if (li.classList.contains('disabled')) {
-						li.remove();
+					ul_dropzone.insertAdjacentHTML('afterbegin', dragHtml);
+					ul_dropzone.firstChild.classList.add('moved');
+					ul_dropzone.scrollTop = 0;
+					const disabledElement = ul_dropzone.getElementsByClassName('disabled');
+					if (ul_dropzone.childElementCount <= 3 && disabledElement.length > 0) {
```

Comment:
What does this mystery number mean exactly? Please either define it above (`const MIN_SOMETHING = 3` for example) or add a comment.

Reply:
it is difficult to explain: the `<ul>` consists of minimum: (1) Add a feed link (2) the message box "Empty category" and (3) the dropped element.
It is used to delete the message box, when the element is dropped to the list.
Any suggestion for a good comment?

BTW, it is not new, just changed the number. See before: https://github.com/mathContao/FreshRSS/blob/edge/p/scripts/category.js#L26-L29

Reply:
It's not new, I added it myself a couple of weeks ago and you overlooked it, lol :-P https://github.com/FreshRSS/FreshRSS/pull/3893

The original does suffer from a lack of explanation as to why the line's there, but 0 is not a mystery number. It just says "if empty". :-) https://github.com/FreshRSS/FreshRSS/commit/ce43b1761b810c89e5e8660e8ae2638e97ca6339

Anyway, now that you've explained what it's for, what about something more along the lines of `if (!p.querySelector('[draggable=true]'))`? That way, much like the 0 thing, you just say if there are no elements that are draggable, without having a mystery number that would need to be updated every time you change something.

Reply:
I have a much more better idea ;)